### ISSUE TITLE
Upgrading webview and local storage

### ISSUE THREAD

**Issue by jairemix** [ionic-team/cordova-plugin-ionic-webview#15]
From my understanding, it seems since UIWebView and WKWebview are different environments, **local storage is not preserved** when switching from one webview to another.

This would be ok for apps not yet in production, but **for apps that already have thousands of customers**, this is a huge compromise. This means that any customer rolling over to the new webview would lose their local storage data and hence lose their settings and login data.

Would there be a way to smoothly transition by transferring local storage data from the old webview to the new, or would my users be forced to go back to a blank slate as they update their app?

**Comment by ghenry22** [comment]
You could release an update still with uiwebview that moves the data into SQLite data Store, this can be accessed from both web views.  Add a migration step into your first release with wk to move the data back to local storage or just keep using sqlite.

**Comment by jairemix** [comment]
Thanks for the suggestion but an intermediate step wasn't really the most ideal solution (since a lot of users are known to skip updates--or even never update).

Instead, I've adapted this plugin to the ionic-webview:

https://github.com/MaKleSoft/cordova-plugin-migrate-localstorage (this was meant to work with cordova-plugin-wkwebview-engine instead).

Here's the link to my solution in case anyone has the same problem migrating localStorage. For good measure, I also decided to migrate IndexedDB files:

https://github.com/jairemix/cordova-plugin-migrate-localstorage
